livedata/src/main/java/io/getstream/chat/android/livedata/EventHandlerImpl.kt

```diff
@@ -95,7 +95,12 @@ class EventHandlerImpl(
         val messagesToFetch = mutableSetOf<String>()
 
         events.filterIsInstance<CidEvent>().onEach { channelsToFetch += it.cid }
-        users += events.filterIsInstance<UserEvent>().map(UserEvent::user).associateBy(User::id)
+        // For some reason backend is not sending us the user instance into some events that they should
+        // and we are not able to identify which event type is. Gson, because it is using reflection,
+        // inject a null instance into property `user` that doesn't allow null values.
+        // This is a workaround, while we identify which event type is, that omit null values without
+        // break our public API
+        users += events.filterIsInstance<UserEvent>().mapNotNull { it.user as User? }.associateBy(User::id)
```

Comment:
Hurts that we have to do a meaningless cast like this, that's logically incorrect given our types, but since our types are incorrect... Anyway, let's at least suppress the warning, so that we don't accidentally remove it 😄 

```suggestion
        @Suppress("USELESS_CAST")
        users += events.filterIsInstance<UserEvent>().mapNotNull { it.user as User? }.associateBy(User::id)
```

Reply:
Yes, I am not happy with this implementation neither. Hope to remove it soon